homework/Repository.js

```diff
@@ -13,8 +13,53 @@ class Repository {
    * @param {HTMLElement} container The container element in which to render the repository.
    */
   render(container) {
-    // TODO: replace the next line with your code.
-    Util.createAndAppend('pre', container, JSON.stringify(this.repository, null, 2));
+    const table = Util.createAndAppend('table', container);
+    const tbody = Util.createAndAppend('tbody', table);
+
+    // 🤷‍ It would be easy if I could use innerHTML property in Util class.
+    // Then I would use template string for '<a>' tag instead of this model.
+    const tableModel = [
+      {
+        title: 'Repository',
+        hasChild: true,
+        value: {
+          tag: 'a',
+          options: {
+            class: '',
+            href: this.repository.html_url,
+            target: '_blank',
+            text: this.repository.name,
+          },
+        },
+      },
+      { title: 'Forks', hasChild: false, value: this.repository.forks || 'No forks' },
+      {
+        title: 'Description',
+        hasChild: false,
+        value: this.repository.description || 'No description',
+      },
+      {
+        title: 'Updated',
+        hasChild: false,
+        value: new Date(this.repository.updated_at).toLocaleString(),
+      },
+    ];
+
+    this.fillTable(tbody, tableModel);
+  }
+
+  fillTable(tbody, model) {
+    model.forEach(row => {
+      const tr = Util.createAndAppend('tr', tbody);
+      if (row.hasChild) {
```

Comment:
Thanks, it's really better approach. 

First I tried if statement with comparing **type** of the child element if it's object or not. But I get some weird errors when I navigate between repositories. I couldn't understand the reason. Then I use hasChild property. With your solutions it makes really more sense.

Reply:
The behaviour you mentioned may have been caused by a (description) field having a `null` value instead of a string for some repository.

```js
console.log(typeof null); // -> 'object'
```

This would, of course, throw the rest of your code off-balance.

Reply:
Yes, exactly. Now I understand why I saw the error. Thank you.